delfin/drivers/security.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import ssl
+
+import requests
+from OpenSSL.crypto import load_certificate, FILETYPE_PEM
+from oslo_config import cfg
+from oslo_log import log
+from urllib3 import PoolManager
+
+from delfin import exception
+
+LOG = log.getLogger(__name__)
+CONF = cfg.CONF
+
+
+class RootRestClient(object):
+    def __init__(self):
+        """
+        Checking the southbound security config validation.
+        As required by requests, ca_path must be a directory prepared using
+        the c_rehash tool included with OpenSSL.
+        Once new certificate added, this function can be called for update.
+        If there is a CA certificate chain, all CA certificates along this
+        chain should be included in a single file.
+        """
+        self.enable_ssl = CONF.southbound_security.enable_ssl
+        self.ca_path = CONF.southbound_security.ca_path
```

Comment:
Can we support specifying a direct CA certificate file, rather than adding CA certificate to the system certificate bundle?

Reply:
Hi @joseph-v Is it necessary to do it like that? Cause in that scenario, we have to manage the mapping between device and ca files, instead of using the ability of exist third party packages.